lib/common/database.dart

```diff
@@ -28,40 +28,59 @@ class DatabaseService {
   }
 
   /// Assign UID. This must be ran before any other database function is called else it will crash
-  /// 
   /// takes the string ID
   initUID(String uid) {
     userid = uid;
   }
 
-  Future<DocumentSnapshot<Map<String, dynamic>>> getEvents(String eventID) {
-    return users
-        .doc(userid)
-        .collection("events")
-        .doc(eventID)
-        .get(); // turn this into a map of eventID to event objects?
+  Future<Event> getEvent(String eventID) async {
+    try {
+      DocumentSnapshot<Map<String, dynamic>> eventDocument =
+          await users.doc(userid).collection('events').doc(eventID).get();
+      if (eventDocument.exists && eventDocument.data() != null) {
+        return Event.fromMap(eventDocument.data() ?? {"getEvent Error": 1},
+            id: eventDocument.id);
+      }
+    } catch (e) {
+      rethrow;
+    }
+    throw Exception("Event not found");
   }
 
-  Future<QuerySnapshot<Map<String, dynamic>>> getAllEvents() {
-    return users
+  Future<List<Event>> getAllEvents() async {
+    List<Event> allEvents = [];
+    final eventDocs = await users.doc(userid).collection("events").get();
+
+    for (var doc in eventDocs.docs) {
+      allEvents.add(Event.fromMap(doc.data(), id: doc.id));
+    }
+
+    return allEvents;
+  }
+
+  /// Put a new event in database. Error if already exists
+  Future<void> addEvent(Event event) async {
```

Comment:
this was moved up from below so better grouping/order

Reply:
Also i changed it to be more similar to setTask but still keeping it must be unique